iceoryx2-cal/tests/resizable_shared_memory_tests.rs

```diff
@@ -0,0 +1,954 @@
+                unsafe { ptr.data_ptr.add(n).write(2 * i as u8) };
+            }
+            ptrs.push(ptr);
+        }
+
+        for i in 0..NUMBER_OF_REALLOCATIONS {
+            let size = 2 + i;
+            let ptr_view = unsafe {
+                sut_viewer
+                    .register_and_translate_offset(ptrs[i].offset)
+                    .unwrap()
+            };
+
+            for n in 0..size {
+                assert_that!(unsafe{ *ptr_view.add(n) }, eq 2*i as u8);
+            }
+        }
+    }
+
+    #[test]
+    fn allocate_with_sufficient_chunk_hint_and_increasing_alignment_strategy_power_of_two<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        allocate_with_sufficient_chunk_hint_and_increasing_alignment::<Shm, Sut>(
+            AllocationStrategy::PowerOfTwo,
+        )
+    }
+
+    #[test]
+    fn allocate_with_sufficient_chunk_hint_and_increasing_alignment_strategy_best_fit<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        allocate_with_sufficient_chunk_hint_and_increasing_alignment::<Shm, Sut>(
+            AllocationStrategy::BestFit,
+        )
+    }
+
+    #[test]
+    fn deallocate_last_segment_does_not_release_it<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        let storage_name = generate_name();
+        let config = generate_isolated_config::<Sut>();
+
+        let sut = Sut::MemoryBuilder::new(&storage_name)
+            .config(&config)
+            .allocation_strategy(AllocationStrategy::Static)
+            .max_chunk_layout_hint(Layout::new::<u8>())
+            .max_number_of_chunks_hint(1)
+            .create()
+            .unwrap();
+
+        assert_that!(sut.allocate(Layout::new::<u8>()), is_ok);
+
+        let result = sut.allocate(Layout::new::<u8>());
+        assert_that!(result, is_err);
+        assert_that!(result.err().unwrap(), eq ResizableShmAllocationError::ShmAllocationError(ShmAllocationError::AllocationError(AllocationError::OutOfMemory)));
+    }
+
+    #[test]
+    fn static_allocation_strategy_does_not_resize_available_chunks<
+        Shm: SharedMemory<DefaultAllocator>,
+        Sut: ResizableSharedMemory<DefaultAllocator, Shm>,
+    >() {
+        let storage_name = generate_name();
+        let config = generate_isolated_config::<Sut>();
+
+        let sut = Sut::MemoryBuilder::new(&storage_name)
+            .config(&config)
+            .allocation_strategy(AllocationStrategy::Static)
+            .max_chunk_layout_hint(Layout::new::<u8>())
+            .max_number_of_chunks_hint(8)
+            .create()
+            .unwrap();
+
+        let result = sut.allocate(Layout::from_size_align(8, 1).unwrap());
+        assert_that!(result, is_err);
+    }
+
+    #[test]
+    fn static_allocation_strategy_increase_available_chunks<
```

Comment:
Is the name correct? Is the static allocation strategy assumed to increase the available chunks?